src/main/java/jenkins/plugins/logstash/persistence/LogzioDao.java

```diff
@@ -0,0 +1,127 @@
+package jenkins.plugins.logstash.persistence;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+import com.google.common.io.Files;
+
+import io.logz.sender.LogzioSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.com.google.gson.JsonObject;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+import jenkins.plugins.logstash.LogstashConfiguration;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Logz.io Data Access Object.
+ *
+ * @author Ido Halevi
+ */
+
+public class LogzioDao extends AbstractLogstashIndexerDao {
+    private static final int CONNECT_TIMEOUT = 10*1000;
+    private static final int CORE_POOL_SIZE = 2;
+    private static final int DRAIN_TIMEOUT = 2;
+    private static final int FS_PERCENT_THRESHOLD = 99;
+    private static final int GC_PERSISTED_QUEUE_FILE_INTERVAL_SECOND = 30;
+    private static final int SOCKET_TIMEOUT = 10*1000;
+    private static final String TYPE = "jenkins_plugin";
+    private final LogzioSender logzioSender;
+
+    private String key;
+    private String host;
+
+    //primary constructor used by indexer factory
+    public LogzioDao(String host, String key){
+        this(null, host, key);
+    }
+
+    // Factored for unit testing
+    LogzioDao(LogzioSender factory, String host, String key){
+        this.host = host;
+        this.key = key;
+
+        // create file for sender queue
+        File fp;
+        Jenkins jenkins = Jenkins.getInstanceOrNull();
+        if (jenkins == null){
+            // for testing (mvn package)
+            fp = Files.createTempDir();
+        }else{
+            fp = new File(jenkins.getRootDir() + "/logs/logzio_plugin");
```

Comment:
I'm not sure if this is the correct location, I think there is some standard location for plugin work-folders but can't find it now. @mwinter69 ?

Who is going to manage the folder to remove stale or orphaned files?
Please add information to the readme about this.

Reply:
The shipper manages it by running a thread that checks every 30s if we can clean the files or not. I can decrease the time if you think 30s is too long

Reply:
I'm not concerned about day-to-day operation.
Are you sure this will manage to cleanup the directory when plugin is uninstalled or if backend is changed to another dao?

Reply:
@jakub-bochenski 
Is there a way to let the dao know that it will be closed soon? Maybe to add it as part of abstraction the dao needs to implement.

Reply:
You are right -- there is no notification now, we will need to add something to `LogstashIndexerDao` then call it from `LogstashConfiguration.configure` etc

Reply:
another thing: with the current way `LogstashConfiguration` works we'd probably need to move this logic out of constructor and into some kind of setup method

Reply:
I'm not sure if it would help in this case. I think it's better to pool threads across the Jenkins instance instead of having a thread pool per job. 
Also this might actually complicate handling re. jobs being aborted, dying unexpectedly etc

Reply:
@jakub-bochenski 
How do you suggest we should progress with this?

Reply:
@jakub-bochenski 
Do you prefer I will change it to in-memory shipper like the rest?

Reply:
@jakub-bochenski  any news?

Reply:
@jakub-bochenski 
Thanks :) I'll try to finish with it today